wagtail/admin/templates/wagtailadmin/base.html

```diff
@@ -27,8 +27,13 @@
                                 {{ message|safe }}
                             </li>
                         {% endfor %}
-                    </ul>
-                {% endif %}
+                    {% endif %}
+                </ul>
+                <template data-w-messages-target="templates">
```

Comment:
After re-reading, I do think discrete templates makes this much easier to reason about in the code, will update in the merge.